src/main.rs

```diff
@@ -9,9 +9,15 @@
 // specific language governing permissions and limitations under the License.
 
 extern crate clap;
+#[macro_use] extern crate nickel;
+
+use nickel::{Nickel, HttpRouter};
 
 use clap::{App, Arg, SubCommand, AppSettings};
 
+use std::u16;
+use std::str::FromStr;
```

Comment:
I also prefer the third approach, feels explicit while also keeping `use` statements readable and easier to maintain over time.